[Tham You Wei] iP

tyouwei opened the pull request:
# **The Evan Chatbot**

>"To die or not to die, that is the question" -Anonymous

Evan frees your mind of having to remember things you need to do. It's,
- text-based (not for the blind)
- easy to learn (at the expense of my sanity)
- design is very human

Want to create a DIY Evan?
1. Apply for NUS CS [here](https://www.join.comp.nus.edu.sg/apply)
2. Make it to the 2nd year
3. Take CS2103T

And it **only costs 5k per semester!!!**

Future update plans:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon 😢)
- [ ] Reminders (coming s00n) 

Also, here's my ``` main ``` method.
```ruby
public static void main(String[] args) {
        //Start user interaction
        Scanner scanner = new Scanner(System.in);

        while(true) {
            try {
                String input = scanner.nextLine().toLowerCase();

                if (input.equals(Command.BYE.getCommand())) {
                    reply.printDialog("Bye. Hope to see you again soon!");
                    return;
                } else if (input.equals(Command.LIST.getCommand())) {
                    tasks.printTasks();
                } else if (input.equals(Command.TODO.getCommand())) {
                    ToDoCommand.start();
                } else if (input.equals(Command.DEADLINE.getCommand())) {
                    DeadlineCommand.start();
                } else if (input.equals(Command.EVENT.getCommand())) {
                    EventCommand.start();
                } else if (input.startsWith(Command.DELETE.getCommand())) {
                    DeleteCommand.start(input);
                } else if (input.startsWith(Command.MARK.getCommand())) {
                    MarkCommand.start(input);
                } else if (input.startsWith(Command.UNMARK.getCommand())) {
                    UnmarkCommand.start(input);
                } else if (input.startsWith(Command.FIND.getCommand())) {
                    FindCommand.start(input);
                } else {
                    throw new InvalidCommandException();
                }
            } catch (InvalidInputException e) {
                reply.printDialog(e.toString());
            } catch (MissingArgumentException e) {
                reply.printDialog(e.toString());
            } catch (InvalidCommandException e) {
                reply.printDialog(e.toString());
            }
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tyouwei committed:
Added Rename, Greet, Exit

tyouwei committed:
Added Echo

tyouwei committed:
Added Add, List

tyouwei committed:
Added Mark as Done

tyouwei committed:
Added ToDos, Events, Deadlines

tyouwei committed:
Added Automated text UI testing and fixed some input bugs

tyouwei committed:
Added error handles

tyouwei committed:
Added delete

tyouwei committed:
Use Enums

tyouwei committed:
Add Save

tyouwei committed:
Organise packages

tyouwei committed:
Add Date Time

tyouwei committed:
Chatbot revamp

tyouwei committed:
Chatbot revamp

tyouwei committed:
Add Time

tyouwei committed:
More OOP

tyouwei committed:
package and cleanup

tyouwei committed:
Merge branch 'branch-level-7' into add-gradle-support

tyouwei committed:
Use Gradle

tyouwei committed:
JUnit

tyouwei committed:
Fix Bug

tyouwei committed:
Delete faulty test

tyouwei committed:
More bug fixes

tyouwei committed:
Add Javadocs

tyouwei committed:
Coding Standard

tyouwei committed:
Add Find

tyouwei committed:
gitignore

tyouwei committed:
gitignore

tyouwei committed:
Command bug fix

tyouwei committed:
add checkstyles

jack1e0 commented on src/main/java/storage/Database.java:
```diff
@@ -0,0 +1,81 @@
+import task.Events;
+import ui.Reply;
+
+import java.io.*;
```
Should import each package separately, instead of *

jack1e0 commented on src/main/java/chatbot/evan/Evan.java:
```diff
@@ -0,0 +1,72 @@
+
+/**
+ * Main class for the chatbot
+ */
```
good javadocs written throughout

jack1e0 commented on src/main/java/parser/Time.java:
```diff
@@ -0,0 +1,175 @@
+     * @return boolean indicating if its the same format or not
+     */
+    private static boolean hasSameDateFormat(String date, DateTimeFormatter formatter) {
+        try {
```
good naming format

jack1e0 commented on src/main/java/enums/Command.java:
```diff
@@ -0,0 +1,45 @@
+    TODO("todo"),
+    EVENT("event"),
+    FIND("find"),
+    SKIP("skip");
```
enum constants are caps 👍

jack1e0 reviewed (commented):
LGTM, coding standards are followed well.

tyouwei committed:
Add GUI

tyouwei committed:
Add GUI

tyouwei committed:
Merge branch 'branch-level-10'

tyouwei committed:
checkstyles

s-peiran commented on src/main/java/chatbot/evan/Evan.java:
```diff
@@ -0,0 +1,80 @@
+import process.SimpleProcess;
+import process.ToDo;
+import process.Unmark;
+import task.TaskList;
```
import statements neatly organized

s-peiran commented on src/main/java/process/Process.java:
```diff
@@ -0,0 +1,14 @@
+     * @return String message of appropriate response for the user input
+     */
+    public String processInput(String input);
+}
```
You may consider to add the undeclared methods in the interface.

s-peiran commented on src/main/java/storage/Database.java:
```diff
@@ -0,0 +1,68 @@
+                list.add(new Deadlines(status, data[2], data[3]));
+            } else if (taskType.equals(Command.EVENT.getCommand())) {
+                list.add(new Events(status, data[2], data[3], data[4]));
+            }
```
good job on writing it neat and non-nested.

s-peiran commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
```
consider returning a new Task object.

s-peiran commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
```
Consider using fstrings/string.format

s-peiran reviewed (commented):
Hi please see some recommendations i have outlined

tyouwei committed:
Add assertions

Added some assertion statements onto all of the task processes for
future debugging.

tyouwei committed:
A-CodeQuality

Improved the code quality to adhere to certain standards and making sure
each function is short and performs a single operation.

tyouwei committed:
A-FullCommitMessage

Successfully commited twice with full commit messages
Also improved CodeQuality for some functions

tyouwei committed:
Bug Fix

Fixed a bug for FileNotFoundException when jar is used

tyouwei committed:
Friendlier Syntax

Made the syntax more user friendly

tyouwei committed:
Test PR merge

tyouwei committed:
Merge pull request #1 from tyouwei/newbranch

Test PR merge

tyouwei committed:
Testing PR

tyouwei committed:
Testing PR

tyouwei committed:
Merge branch 'master' into branch-A-Assertions

tyouwei committed:
Merge pull request #3 from tyouwei/branch-A-Assertions

Merging PRs

tyouwei committed:
Merge pull request #2 from tyouwei/branch-A-CodeQuality

Merging PRs

tyouwei committed:
Add files via upload

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
Update README.md

tyouwei committed:
A-UserGuide

tyouwei committed:
A-UserGuide

tyouwei committed:
A-UserGuide

Add a comprehensive user guide for Evan Chatbot

tyouwei committed:
README

Updated the README file

tyouwei committed:
Updated main README

updated the main README to remove ny references to Duke

tyouwei committed:
Changed save file location

Changed the save file location to the parent directory of the jar file

tyouwei committed:
More JUnit Testing

Added more JUnit Tests for Deadline process

tyouwei committed:
Add images

Added user and bot image

tyouwei committed:
Update Ui.png

tyouwei committed:
Update Ui.png
